opendevin/runtime/docker/image_agnostic_util.py

```diff
@@ -0,0 +1,90 @@
+import tempfile
+
+import docker
+
+from opendevin.core.logger import opendevin_logger as logger
+
+
+def generate_dockerfile_content(base_image: str) -> str:
+    """
+    Generate the Dockerfile content for the agnostic sandbox image based on user-provided base image.
+
+    NOTE: This is only tested on debian yet.
+    """
+    # FIXME: Remove the requirement of ssh in future version
+    dockerfile_content = (
+        f'FROM {base_image}\n'
+        'RUN apt update && apt install -y openssh-server wget sudo\n'
+        'RUN mkdir -p -m0755 /var/run/sshd\n'
+        'RUN mkdir -p /opendevin && mkdir -p /opendevin/logs && chmod 777 /opendevin/logs\n'
+        'RUN wget "https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-$(uname)-$(uname -m).sh"\n'
+        'RUN bash Miniforge3-$(uname)-$(uname -m).sh -b -p /opendevin/miniforge3\n'
+        'RUN bash -c ". /opendevin/miniforge3/etc/profile.d/conda.sh && conda config --set changeps1 False && conda config --append channels conda-forge"\n'
+        'RUN echo "export PATH=/opendevin/miniforge3/bin:$PATH" >> ~/.bashrc\n'
+    ).strip()
+    return dockerfile_content
+
+
+def _build_sandbox_image(
+    base_image: str, target_image_name: str, docker_client: docker.DockerClient
+):
+    try:
+        with tempfile.TemporaryDirectory() as temp_dir:
+            dockerfile_content = generate_dockerfile_content(base_image)
+            logger.info(f'Building agnostic sandbox image: {target_image_name}')
+            logger.info(
+                (
+                    f'===== Dockerfile content =====\n'
+                    f'{dockerfile_content}\n'
+                    f'==============================='
+                )
+            )
+            with open(f'{temp_dir}/Dockerfile', 'w') as file:
+                file.write(dockerfile_content)
+
+            image, logs = docker_client.images.build(
+                path=temp_dir, tag=target_image_name
+            )
+
+        for log in logs:
+            if 'stream' in log:
+                print(log['stream'].strip())
+
+        logger.info(f'Image {image} built successfully')
+    except docker.errors.BuildError as e:
+        logger.error(f'Sandbox image build failed: {e}')
+        raise e
+    except Exception as e:
+        logger.error(f'An error occurred during sandbox image build: {e}')
+        raise e
+
+
+def _get_new_image_name(base_image: str) -> str:
+    return base_image + '_od_sandbox'
```

Comment:
```plaintext
00:27:21 - opendevin:INFO: image_agnostic_util.py:55 - Image <Image: 'opendevin_agnostic_sandbox:latest', 'ubuntu:22.04_od_sandbox'> built successfully
```

What if we use tags for differentiation? Like this: `od_sandbox:ubuntu_22.04`, `od_sandbox:debian_11`. Would this be better than `ubuntu:22.04_od_sandbox`? Perhaps this looks cleaner?

Reply:
`od_sandbox:ubuntu_22.04` sounds better than `ubuntu:22.04_od_sandbox` to me